test/q-transformer.test.js

```diff
@@ -4846,4 +4852,203 @@ describe('qTransformer', () => {
+                            pageTitle: 'Email address',
+                            hasErrors: false,
+                            content: removeIndentation(`
+                        {% from "input/macro.njk" import govukInput %}
+                        {{ govukInput({
+                            "id": "email",
+                            "name": "email",
+                            "type": "email",
+                            "label": {
+                                "html": "Email address",
+                                "isPageHeading": true,
+                                "classes": "govuk-label--xl"
+                            },
+                            "hint": {
+                                "text": "(Optional)"
+                            }
+                        }) }}`)
+                        };
+                        expect({
+                            pageTitle: result.pageTitle,
+                            hasErrors: result.hasErrors,
+                            content: removeIndentation(result.content)
+                        }).toEqual(expected);
+                    });
+                });
+            });
+            describe('Property is not a page heading', () => {
+                it('Should append " (Optional)" on title', () => {
+                    const result = qTransformer.transform({
+                        schemaKey: 'event-name',
+                        schema: {
+                            type: 'object',
+                            properties: {
+                                email: {
+                                    type: 'string',
+                                    description: 'e.g. something@something.com',
+                                    format: 'email',
+                                    title: 'Email address'
+                                },
+                                phone: {
+                                    type: 'string',
+                                    description: 'e.g. 07799111111',
+                                    title: 'Phone'
+                                }
+                            }
+                        },
+                        uiSchema: {}
+                    });
+
+                    const expected = {
+                        pageTitle: 'Email address',
+                        hasErrors: false,
+                        content: removeIndentation(`
+                        {% from "input/macro.njk" import govukInput %}
+                        {{ govukInput({
+                            "id": "email",
+                            "name": "email",
+                            "type": "email",
+                            "label": {
+                                "html": "Email address",
+                                "isPageHeading": true,
+                                "classes": "govuk-label--xl"
+                            },
+                            "hint": {
```

Comment:
is this not the equivalent of the schema `description` and not the `title`? Just asking because the title of this test is `it('Should append " (Optional)" on title', () => {`

Reply:
This would be expected as the first element here is a page heading and should have the string appended to its description. The functionality I was testing was demonstrated below in the second element which was not a heading and so should have had the string appended to it's title. However I have modified this test and added a page-level title to remove this ambiguity. Now both elements should have the string appended to their title as neither is a page heading.